INGRID/geometry.py

```diff
@@ -976,43 +1012,257 @@ def find_nearest(array, value):
+                            has_limit=self.distortion_correction['has_limit'], du_min=min_du,
+                            MaxTol=self.distortion_correction['MaxTol'],
+                            MinTol=self.distortion_correction['MinTol'])
+
+        else:
+            print("distortion correction : reverse", "has_limit",
+                  self.distortion_correction['has_limit'],
+                  self.distortion_correction["du_min"], self.distortion_correction["MaxTol"])
+
+            for i in reversed(range(len(self.W_vertices) - 2)):
+                print("i=", i, len(self.W_vertices) - 2)
+                for j in range(np_lines):
+                    u = _poloidal_f(j / (np_lines - 1))
+                    if self.distortion_correction['active'] and j > 0 and j < np_lines - 1:
+                        Res = self.distortion_correction['resolution']
+                        ThetaMin = self.distortion_correction['theta_min']
+                        ThetaMax = self.distortion_correction['theta_max']
+                        umin = _poloidal_f((j - 1) / (np_lines - 1))
+                        umax = _poloidal_f((j + 1) / (np_lines - 1))
+                        Pt1 = radial_vertices[i+2][j]
+                        Pt2 = radial_vertices[i+2][j - 1]
+                        Tag = '---- Correcting points: {},{}'.format(i, j)
+                        Pt = radial_vertices[i+1][j]
+                        if j == (np_lines - 2) or j == 1:
+                            min_du = self.distortion_correction["du_min"]
+                            if j == 1:
+                                plt.plot(Pt.x, Pt.y, '.', color='red',
+                                         markersize=8, marker='d')
+                                plt.show()
+                                plt.draw()
+                                u_du_min = umin
+                            else:
+                                u_du_min = umax
+                                plt.plot(Pt.x, Pt.y, '.', color='blue',
+                                         markersize=8, marker='d')
+                                plt.show()
+                                plt.draw()
+                        else:
+                            min_du = -10.0
+                            u_du_min = 10.0
+                        Pt = CorrectDistortion(
+                            u, Pt, Pt1, Pt2, self.radial_spl[
+                                i], ThetaMin, ThetaMax, umin, umax, u_du_min, Res, visual, Tag, Verbose=verbose,
+                            has_limit=self.distortion_correction[
+                                'has_limit'], du_min=min_du,
+                            MaxTol=self.distortion_correction['MaxTol'],
+                            MinTol=self.distortion_correction['MinTol']
+                        )
+                        radial_vertices[i+1][j] = Pt
+
+            if self.distortion_correction['active']:
+                for j in range(1, np_lines-1):
```

Comment:
Following up the description request, is this suggesting that we are still maintaining the old code capabilities of `distortion_correction`?

Reply:
Yes. This is just improvement through optional keywords. The default capabilities is untouched (but need to be validated due to refactoring of the correction implementation)